### ISSUE TITLE
CaseClassSigParser#loadClass fails for external jars (Maven vs IDE)

### ISSUE THREAD

**Issue by astubbs** [codahale/jerkson#35]
My stuff works fine in IDE, but when running from Maven, my tests fail:

```
 [main] ERROR c.s.j.s.container.ContainerResponse       - The exception contained within MappableContainerException could not be mapped to a response, re-throwing to the HTTP container
  java.lang.ClassNotFoundException: com.foo.bar.Bottle
  at java.net.URLClassLoader$1.run(URLClassLoader.java:202) ~[na:1.6.0_26]
  at java.security.AccessController.doPrivileged(Native Method) ~[na:1.6.0_26]
  at java.net.URLClassLoader.findClass(URLClassLoader.java:190) ~[na:1.6.0_26]
  at java.lang.ClassLoader.loadClass(ClassLoader.java:306) ~[na:1.6.0_26]
  at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301) ~[na:1.6.0_26]
  at java.lang.ClassLoader.loadClass(ClassLoader.java:247) ~[na:1.6.0_26]
  at com.codahale.jerkson.util.CaseClassSigParser$.loadClass(CaseClassSigParser.scala:135) ~[jerkson_2.9.1-0.4.2.jar:na]
  at com.codahale.jerkson.util.CaseClassSigParser$.findRootClass(CaseClassSigParser.scala:49) ~[jerkson_2.9.1-0.4.2.jar:na]
  at com.codahale.jerkson.util.CaseClassSigParser$.parseScalaSig(CaseClassSigParser.scala:42) ~[jerkson_2.9.1-0.4.2.jar:na]
  at com.codahale.jerkson.util.CaseClassSigParser$.findSym(CaseClassSigParser.scala:56) ~[jerkson_2.9.1-0.4.2.jar:na]
  at com.codahale.jerkson.util.CaseClassSigParser$.parse(CaseClassSigParser.scala:83) ~[jerkson_2.9.1-0.4.2.jar:na]
```

Works fine in IDE (dependent jar containing the class is a part of the modular project). Class loading issue?
I can construct an instance of the class (Bottle) just fine at the beginning of the unit test.

**Comment by astubbs** [comment]
Or i suppose it could be something to do with the classloader setup used for running maven tests...

**Comment by astubbs** [comment]
Turns out its related to surefire's class loading system.
Downgrading to 2.8.1 fixes it - they changed class loading in 2.9
see https://github.com/wicketstuff/core/issues/50
